terraform/main.tf

```diff
@@ -0,0 +1,104 @@
+terraform {
+ required_providers {
+ azurerm = {
+  source = "hashicorp/azurerm"
+  version = ">= 3.8"
+  }
+ }
+ backend "azurerm" {
+        resource_group_name  = "tfstate"
+        storage_account_name = "module12asa"
+        container_name       = "tfstate"
+        key                  = "terraform.tfstate"
+    }
+}
+
+provider "azurerm" {
+ features {}
+}
+
+# azure resource group
+data "azurerm_resource_group" "main" {
+ name = "Cohort22_PauCur_ProjectExercise"
+}
+
+# app service plan
+resource "azurerm_service_plan" "main" {
+  name                = "${var.prefix}-terraformed-asp"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  os_type             = "Linux"
+  sku_name            = "B1"
+}
+
+# azure webapp
+resource "azurerm_linux_web_app" "main" {
+  name                = "app-module12-pdc"
+  location            = data.azurerm_resource_group.main.location
+  resource_group_name = data.azurerm_resource_group.main.name
+  service_plan_id     = azurerm_service_plan.main.id
+
+  site_config {
+    application_stack {
+      docker_image     = "paulcurren/todo-app"
+      docker_image_tag = "prod"
+    }
+  }
+
+  app_settings = {
+    "DOCKER_REGISTRY_SERVER_URL" = "https://index.docker.io"
+    "FLASK_APP"                  = "todo_app/app:create_app()"
+    "FLASK_ENV"                  = "production"
+    "SECRET_KEY"                 = "secret-key"
```

Comment:
Minor for our app but worth noting that having this be public does invalidate all of our security as people can just change their own cookies to say they're signed in as another user so we should really be treating this as sensitive like the other secrets